### ISSUE TITLE
change anchor order influence train ?

### ISSUE THREAD

**Issue by yja1** [zisianw/FaceBoxes.PyTorch#77]
I want to change anchor order in prior_box.py: it influence train?
now train loss not convergence, don't know why

class PriorBox(object):
    def __init__(self, cfg, box_dimension=None, image_size=None, phase='train'):
        super(PriorBox, self).__init__()
        self.variance = cfg['variance']
        self.min_sizes = cfg['min_sizes']
        self.steps = cfg['steps']
        self.aspect_ratios = cfg['aspect_ratios']
        self.clip = cfg['clip']
        if phase == 'train':
            self.image_size = (cfg['min_dim'], cfg['min_dim'])
            self.feature_maps = cfg['feature_maps']
        elif phase == 'test':
            self.feature_maps = box_dimension.cpu().numpy().astype(np.int)
            self.image_size = image_size
        for v in self.variance:
            if v <= 0:
                raise ValueError('Variances must be greater than 0')

    def forward(self):
        mean = []
        for k, f in enumerate([[32, 32]]):
            min_sizes = [32]
            for i in range(f[0]):
                for i in [i+0.125, i+0.375, i+0.625, i+0.875]:
                    for j in range(f[1]):
                        for min_size in min_sizes:
                            s_kx = min_size / self.image_size[1]
                            s_ky = min_size / self.image_size[0]
                            if min_size == 32:
                                dense_cx = [x*self.steps[k]/self.image_size[1] for x in [j+0.125, j+0.375, j+0.625, j+0.875]]
                                dense_cy = [y*self.steps[k]/self.image_size[0] for y in [i]]

                                for cy, cx in product(dense_cy, dense_cx):
                                    mean += [cx, cy, s_kx, s_ky]

        for k, f in enumerate([[32,32]]):
            min_sizes = [64]
            for i in range(f[0]):
                for i in [i+0.25, i+0.75]:
                    for j in range(f[1]):
                        for min_size in min_sizes:
                            s_kx = min_size / self.image_size[1]
                            s_ky = min_size / self.image_size[0]
                            if min_size == 64:
                                dense_cx = [x*self.steps[k]/self.image_size[1] for x in [j+0.25, j+0.75]]
                                dense_cy = [y*self.steps[k]/self.image_size[0] for y in [i]]
                                for cy, cx in product(dense_cy, dense_cx):
                                    mean += [cx, cy, s_kx, s_ky]

        for k, f in enumerate([[32,32]]):
            min_sizes = [128]
            for i in range(f[0]):
                for j in range(f[1]):
                    for min_size in min_sizes:
                        s_kx = min_size / self.image_size[1]
                        s_ky = min_size / self.image_size[0]
                        if min_size == 128:
                            cx = (j + 0.5) * self.steps[k] / self.image_size[1]
                            cy = (i + 0.5) * self.steps[k] / self.image_size[0]
                            mean += [cx, cy, s_kx, s_ky]

        for k, f in enumerate([[16,16]]):
            min_sizes = [256]
            for i in range(f[0]):
                for j in range(f[1]):
                    for min_size in min_sizes:
                        s_kx = min_size / self.image_size[1]
                        s_ky = min_size / self.image_size[0]
                        if min_size == 256:
                            cx = (j + 0.5) * self.steps[k] / self.image_size[1]
                            cy = (i + 0.5) * self.steps[k] / self.image_size[0]
                            mean += [cx, cy, s_kx, s_ky]

        for k, f in enumerate([[8,8]]):
            min_sizes = [512]
            for i in range(f[0]):
                for j in range(f[1]):
                    for min_size in min_sizes:
                        s_kx = min_size / self.image_size[1]
                        s_ky = min_size / self.image_size[0]
                        if min_size == 512:
                            cx = (j + 0.5) * self.steps[k] / self.image_size[1]
                            cy = (i + 0.5) * self.steps[k] / self.image_size[0]
                            mean += [cx, cy, s_kx, s_ky]
        # back to torch land
        output = torch.Tensor(mean).view(-1, 4)
        if self.clip:
            output.clamp_(max=1, min=0)
        return output
